src/contexts/ModalContext.tsx

```diff
@@ -0,0 +1,38 @@
+'use client'
+
+import { PropsWithChildren, createContext, useContext, useState } from 'react'
+
+export type ModalContextType = {
+  profileId: string
+  setProfileId: (id: string) => void
+  showRejectModal: boolean
+  setShowRejectModal: (state: boolean) => void
+}
+
+const ModalContext = createContext<ModalContextType | undefined>(undefined)
+
+export function ModalProvider({ children }: PropsWithChildren) {
+  const [showRejectModal, setShowRejectModal] = useState(false)
+  const [profileId, setProfileId] = useState('')
+
+  return (
+    <ModalContext.Provider
+      value={{
+        profileId,
+        setProfileId,
+        showRejectModal,
+        setShowRejectModal,
+      }}
+    >
+      {children}
```

Comment:
**nitpick:** I would suggest considering using React.portal here to render modal. It will prevent uneccessary rerenders